index.js

```diff
@@ -416,6 +431,32 @@ Primus.readable('forEach', function forEach(fn, done) {
 });
 
 /**
+ * Send a ping packet to all clients to ensure that they are still connected.
+ *
+ * @returns {Primus}
+ * @api private
+ */
+Primus.readable('heartbeat', function heartbeat() {
+  this.forEach(function forEach(spark) {
+    if (!spark.alive) {
```

Comment:
When heartbeats are enabled there is (by default) also a timer on the client that will close it if no ping is received.

Is this additional complexity really needed? If the client uses Primus there are no problems. If not, protocol level pings can be sent instead. Isn't this sufficient?